packages/apollo-link-error/src/index.ts

```diff
@@ -1,9 +1,10 @@
 import { ApolloLink, Observable, Operation } from 'apollo-link';
 import { GraphQLError, ExecutionResult } from 'graphql';
+import { ServerError } from 'apollo-link-http-common';
 
 export interface ErrorResponse {
   graphQLErrors?: GraphQLError[];
-  networkError?: Error;
+  networkError?: ServerError;
```

Comment:
Let's make this a union type of Error, ServerError, and [ServerParseError](https://github.com/apollographql/apollo-link/blob/master/packages/apollo-link-http-common/src/index.ts#L21), since it really could be all three for now. Something like `Error | ServerError | ServerParseError`.

Reply:
When I read your explanation, I think the union type(`Error | ServerError | ServerParseError`) is good.

Reply:
Great point! I'm not sure the best option. It looks like one option is to change the [`name`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/name) field on the networkError to distinguish the different kinds. I think it is acceptable to add a switch on the networkError's name, since it shouldn't happen in your critical network path. If name doesn't work then we could add another field to distinguish them. An other option is to cast it directly in `(networkError as ServerError).statusCode`, though that's super non-ideal. A third option is to change the error interfaces into classes and use `instanceof`, which has a bunch of gotchas. I'd love to hear your thoughts or if you have a better way forward.

If we don't see a better alternative and choose go down the path of changing `name`, then the places to add to are [here](https://github.com/apollographql/apollo-link/blob/master/packages/apollo-link-http-common/src/index.ts#L112), [here](https://github.com/apollographql/apollo-link/blob/master/packages/apollo-link-http-common/src/index.ts#L131), and [here](https://github.com/apollographql/apollo-link/blob/master/packages/apollo-link-http-common/src/index.ts#L249). I think the interfaces of the different errors have to change to include name as well.